### PR TITLE
Drupal with Integrated Composer

### PR DESCRIPTION
**Pull request by EdwardAngert**
## Summary

**[Integrated Composer](https://pantheon.io/docs/integrated-composer)** - Introduce customers to Limited Availability program for Drupal 9 with integrated Composer. [PR 5663](https://github.com/pantheon-systems/documentation/pull/5663)

## Effect

The following changes are already committed:

- new doc walks users through using the integrated composer custom upstream
- Drupal 9 doc updated to incorporate Integrated Composer site creation workflow and LA status.

## Remaining Work

The following changes still need to be completed:

- [x] Tech edit from @ari-gold 
- [x] Copy edit

## Post Launch

**Do not remove** - To be completed by the docs team upon merge:

- [ ] Include/exclude pages ^ respectively within docs search service provider (if applicable)
- [ ] Update Status Report
- [ ] Remove from the [project board](https://github.com/pantheon-systems/documentation/projects/14)

### CODE REVIEW

**Comment by pantheondocs** [comment]
Lighthouse Audit Results:

| Description | Live | pr-5663 | Difference |
| --- | --- | --- | --- |
| Performance Score | `56.00000000000001` | `44` | `-12` |
| Best Practice Score | `86` | `86` | `0` |
| SEO Score | `96` | `96` | `0` |
| Accessibility Score | `96` | `96` | `0` |

Lighthouse Audit Reports:

- [`pr-5663` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/20301/artifacts/0//lighthouse_data/lighthouse-audit-dev.html)
- [`Pantheon Docs` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/20301/artifacts/0//lighthouse_data/lighthouse-audit-reference.html)

File: source/content/drupal-integrated-composer.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
+---
+title: Integrated Composer Early Access
```
Comment (ari-gold) [inline source/content/drupal-integrated-composer.md:2]: Limited Availability

**Comment by mbaynton** [comment]
These docs look good as a quick start for using Integrated Composer with our specific starter template.

I think there should also be general documentation for people who want to let Pantheon run Composer with their existing Composerized Drupal or WordPress site. There's more things that users need to know to be successful on this path, though. A big one is that we strictly require any file created by Composer to be ignored by git via the repository's .gitignore. This includes `/vendor`, but on a Drupal 8+ site it also includes many other paths as well. So the user is faced with the task of writing a correct .gitignore. We should probably provide / link to in documentation a reference .gitignore that covers the typical paths that Composer installs files to for a Drupal site. The .gitignore in https://github.com/pantheon-upstreams/drupal-project/blob/master/.gitignore should fit the bill.

**Comment by ari-gold** [comment]
Please hold on merging this doc. Would like to finalize a few more product features before entering Limited Availability

File: source/content/drupal-integrated-composer.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+contributors: [edwardangert]
+searchboost: 150
+reviewed: "2020-07-21"
+---
+
+Pantheon is testing a release of Drupal with integrated Composer and is opening the feature to users for Limited Availability testing. Integrated Composer lets you deploy your site on Pantheon with one-click updates for both upstream commits and Composer dependencies, while still receiving upstream updates.
+
+## Get access to Integrated Composer
+
+Contact [Support](/support). If you are not part of an organization already, your CSE will add you to a test organization from which you can deploy the upstream. If you are already part of an organization, the feature can be enabled organization-wide.
```
Comment (ari-gold) [inline source/content/drupal-integrated-composer.md:15]: Integrated Composer is currently limited to customers with a Pantheon organization. Contact [Support](/support) to request access. Once enabled for your organization, you can enable it for a custom upstream via pantheon.upstream.yml or for individual sites via pantheon.yml.

Comment (EdwardAngert) [reply source/content/drupal-integrated-composer.md]: updated to match the first part
I'd like to include the second part in the next section, would it be a better step 3 there?
> enable it for a custom upstream via pantheon.upstream.yml or for individual sites via pantheon.yml

File: source/content/integrated-composer.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+├─ README.md
+└─ upstream-config/
+   └─ composer.json
+```
+
+- `.gitignore`: Prevents build artifacts generated by Composer from being committed to the upstream or site code repositories.
+- `composer.json`: The two different `composer.json` files allow customization of individual sites without inherent merge conflicts and enable one-click updates.
+  - Root-level: Site-level customizations.
+  - `upstream-config/composer.json`: Customizations for the upstream
+- `pantheon.upstream.yml`: The `build_step_demo: true` directive in `pantheon.upstream.yml` enables the build step. This name is temporary and will change.
```
Comment (alexdicianu) [inline source/content/integrated-composer.md:55]: `build_step_demo` has been renamed to `build_step`

File: source/content/integrated-composer.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+When a site is created, Pantheon runs `composer install`, generates a `composer.lock` file and commits it back to the site’s code repository.
+
+Build artifacts are stored in a Git tag (`pantheon_build_artifacts_master`), not the main branch (`master` for Dev or a Multidev feature branch).
```
Comment (alexdicianu) [inline source/content/integrated-composer.md:59]: If we intend to go into the implementation details, there's a caveat here in that the tag name is dependent upon the multidev env, thus its name is: `pantheon_build_artifacts_{BRANCH_NAME}`

File: source/content/integrated-composer.md [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+  cd upstream-config
+  ```
+
+1. Run:
+
+  ```bash{promptUser: user}
+  composer require drupal/pkg-name --no-update
+  ```
+
+   - `--no-update` makes Composer run faster
```
Comment (alexdicianu) [inline source/content/integrated-composer.md:77]: That's true, it also disables the automatic updates of dependencies which might be good to also specify.

File: source/content/integrated-composer.md [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+  ```
+
+1. Commit `composer.json` and `composer.lock` and push.
+
+   - Pantheon will run Composer, generate build artifacts, and deploy it to your Dev or Multidev environment.
+
+1. Remove dependencies:
+
+  ```bash{promptUser: user}
+  composer remove
```
Comment (alexdicianu) [inline source/content/integrated-composer.md:123]: Simply running `composer remove` is not going to work. We need to tell it which packages to remove.

File: source/content/integrated-composer.md [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+
+The upstream has the following directory structure:
+
+```none:title=core/
+core/
+├─ .gitignore
+├─ composer.json
+└─ pantheon.upstream.yml
+├─ README.md
+└─ upstream-config/
```
Comment (alexdicianu) [inline source/content/integrated-composer.md:47]: I think this directory was changed to `upstream-configuration`

**Comment by dficker** [comment]
I also noticed that https://pr-5663--pantheon-docs.my.pantheonfrontend.website/docs/integrated-composer does mention that the pantheon.upstream.yml has the "build_step_demo: true" as a part of that file.

I see that the pantheon.upstream.yml has the directive "build_step: true". So the Docs needs to be updated with that.

File: source/content/integrated-composer.md (hunk 4 above)
Comment (EdwardAngert) [reply source/content/integrated-composer.md]: thanks! I adjusted [L61](https://github.com/pantheon-systems/documentation/pull/5663/commits/90f5dfddf836f76060f647d6f37ae69258e32e13#diff-c609f8474a4e653f4ee707364d61540ac160cd501467d72eec5663504401d3ddR61) to try to explain this

File: source/content/integrated-composer.md (hunk 5 above)
Comment (EdwardAngert) [reply source/content/integrated-composer.md]: explained in [L79](https://github.com/pantheon-systems/documentation/pull/5663/commits/90f5dfddf836f76060f647d6f37ae69258e32e13#diff-c609f8474a4e653f4ee707364d61540ac160cd501467d72eec5663504401d3ddR79)

**Comment by EdwardAngert** [comment]
Removed the Drupal 9 Build Tools section for now and started a draft PR that puts it back for when it's time: https://github.com/pantheon-systems/documentation/pull/6125

File: source/content/drupal-9.md [hunk 8]
```diff
@@ -13,19 +13,17 @@ Drupal 9 is, according to [drupal.org](https://www.drupal.org/docs/understanding
 
 Drupal 9 updates Drupal’s underlying dependencies like [Symfony 4.4](https://symfony.com/releases/4.4) and [Twig 2](https://twig.symfony.com/doc/2.x/index.html), removes several deprecated API functions in favor of better options, and allows everyone running Drupal 8.8+ an easy upgrade path to Drupal 9 and beyond.
```
Comment (ari-gold) [inline source/content/drupal-9.md:14]: While Drupal 9.0 was all about removing deprecations, Drupal 9.1 has "new cutting-edge features" and will be supported beyond the Nov 2021 Drupal 8 end-of-life. So I think this intro should be less measured and more celebratory of Drupal 9. Maybe take some cues from https://www.drupal.org/about/9 or https://www.drupal.org/about/media-kit/drupal-9  -- maybe @mmlc or @stevector can help too?

**Comment by alexdicianu** [comment]
hey @ari-gold @EdwardAngert are there any blockers to this article getting published? Or are just waiting on an official release?

I think we're going to include very minimal documentation in the drupal/wordpress upstream repos themselves and just point to this page.

File: source/content/drupal-9.md [hunk 9]
```diff
@@ -4,67 +4,26 @@ description: Test the latest Drupal on the Pantheon WebOps Platform.
...
-
-1. Use Terminus to create a new Drupal 9 site, complete with a customizable CI workflow. Remember to replace `<NEW-SITE-NAME>` in this example with a name for your new site, and `optional-pantheon-organization-name` with your choice (if any):
+If you're not ready to create a new site yet, you can also [check an existing site's compatibility to upgrade](#test-an-existing-drupal-site-for-drupal-9-upgrade-compatibility). Once you're ready, [test-upgrade an existing Pantheon Drupal 8 site to Drupal 9](#test-upgrade-an-existing-pantheon-drupal-8-site-to-drupal-9).
 
-  ```bash{promptUser: user}
-  terminus build:project:create "pantheon-systems/drupal-9-with-circleci-orb:dev-master" <NEW-SITE-MACHINE-NAME> --stability=dev --team='optional-pantheon-organization-name'
-  ```
+<Partial file="drupal-9-upstream-install.md" />
 
 ## Test an Existing Drupal Site for Drupal 9 Upgrade Compatibility
```
Comment (mbaynton) [inline source/content/drupal-9.md:28]: All existing D8 sites will be running MariaDB 10.0, and to run D9 they'll need to be upgraded to 10.4. What I'm hearing from @Hornswoggles is that we're not ready to update existing sites across db server versions yet, so I'm not sure we can publish instructions about upgrading existing sites at this time.

File: source/partials/drupal-9-upstream-install.md [hunk 10]
```diff
@@ -0,0 +1,15 @@
+1. Log in to your Pantheon account. If you don't have one, [create one first](https://pantheon.io/register?docs). Familiarize with the [User Dashboard](/guides/quickstart/user-dashboard) and return to this section. You'll create a new site in the next step.
+
+1. Use this link to create a new Drupal 9 Integrated Composer site from the Upstream: [https://dashboard.pantheon.io/sites/create?upstream_id=56e0a55f-fa74-49d2-a4b3-af18dacee008](https://dashboard.pantheon.io/sites/create?upstream_id=56e0a55f-fa74-49d2-a4b3-af18dacee008). On the **Create Your Pantheon Site** page:
```
Comment (mbaynton) [inline source/partials/drupal-9-upstream-install.md:3]: Are we adding D9 to the Create Site page @ari-gold?

File: source/content/integrated-composer.md [hunk 11]
```diff
@@ -0,0 +1,162 @@
...
+
+### Drupal 9 with Integrated Composer
+
+Please note the Limited Availability program does not include a path to upgrade from previous Drupal versions to Drupal 9. Upgrade instructions for existing Drupal 8 Composer-enabled sites will be available when Integrated Composer moves into General Availability.
+
+<Partial file="drupal-9-upstream-install.md" />
+
+### WordPress with Integrated Composer
+
+1. [Fork the Pantheon-maintained repository](/create-custom-upstream#create-and-host-the-repository-remotely) from [https://github.com/pantheon-upstreams/wordpress-project](https://github.com/pantheon-upstreams/wordpress-project).
```
Comment (mbaynton) [inline source/content/integrated-composer.md:24]: Any reason not to use a `https://dashboard.pantheon.io/sites/create?upstream_id=` -style URL for WordPress with IC, from a product point of view? Making your own fork and custom upstream isn't a requirement technically.

File: source/partials/drupal-9-upstream-install.md (hunk 10 above)
Comment (ari-gold) [inline source/partials/drupal-9-upstream-install.md:3]: updated link: https://dashboard.pantheon.io/sites/create?upstream_id=e96c6794-77fe-4931-9a20-48a2fe1a3789

Comment (carolyn-shannon) [reply source/partials/drupal-9-upstream-install.md]: done

**Review by ari-gold — APPROVED** [review]

**Pull request merged**